### ISSUE TITLE
States are not documented and possibly redundant

### ISSUE THREAD

**Issue by mturilli** [radical-cybertools/radical.pilot#2427]
Events are documented at: https://github.com/radical-cybertools/radical.pilot/blob/devel/docs/source/events.md but states documented at https://github.com/radical-cybertools/radical.pilot/wiki/State-Model-Evolution are marked at OUTDATED. Thus, we do not have an authoritative documentation of RP states. 

Should we just drop the whole notion of state and use only that of event? The current states would be marked as guaranteed to fire in the documentation---i.e., states in the current narrow radical-specific definition---but that distinction would not percolate in RCT notation. This would greatly simply the use of events in RA, both conceptually and syntactically. 

Distinguishing between states and events is redundant. RA returns an empty list if a specific event is not present, either via `re.entity.timestamps()` or `ra.entity.duration()` making the syntactic distinction between events and state cumbersome and clunky. We have to represent an event as: `{1: 'name_of_event', 5: None}` and a state as `{1: 'state', 5: 'NAME_OF_STATE'}`, when we could just represent them as 'name_of_event' and 'NAME_OF_STATE', without using a dict or the values `None` and `state`.

**Comment by andre-merzky** [comment]
I tend to agree that the notion of states is not useful on the level of RA, and also agree that it would simplify the RA API and RA's use in general if we exclusively focus on the notion of events.  The states do serve a well defined purpose in RP though, as they define the life cycle of task and pilot entities and orchestrate the communication of those entities between RP components (and similarly in RE and RS, too) - but that orchestration does not need to (directly) be tied to performance events / traces.
